### PR TITLE
Add functions to support datamover backup flow

### PR DESCRIPTION
**Pull request by sbahar619**
I AddED functions to support DataMover backup flow.

### CODE REVIEW

File: src/oadp_resources/velero/backup.py [hunk 1]
```diff
@@ -131,3 +135,11 @@ def wait_for_deleting(self, wait_timeout=200, sleep=1):
             sleep=sleep,
             wait_timeout=wait_timeout
         )
+
+    def wait_for_done(self, wait_timeout=200, sleep=1):
+        return wait_for(
+            condition_function=self.done,
+            description=f"{self.kind} backup to done, {self.name}",
```
Comment (amastbau) [inline src/oadp_resources/velero/backup.py:142]: backup to be done Or the Backup process to finish.
wdyt?

**Review by amastbau — CHANGES_REQUESTED** [review]

File: src/oadp_resources/velero/backup.py (hunk 1 above)
Comment (sbahar619) [reply src/oadp_resources/velero/backup.py:142]: Finish, dine, is the same... nope?

File: src/oadp_resources/oadp/datamover/volume_snapshot_backup.py [hunk 2]
```diff
@@ -76,22 +89,23 @@ def get_by_source_pvc(cls, src_pvc_name: str, vsb_list: list = None):
 
         vsbl_size = len(vsb_filtered_list)
         if vsbl_size > 1:
-            logger.error(f"More than one VSB was found with source PVC name {src_pvc_name}")
+            logger.info(f"More than one VSB was found with source PVC name {src_pvc_name}")
 
         if vsbl_size == 0:
-            logger.error(f"No VSB was found with source PVC name {src_pvc_name}")
+            logger.info(f"No VSB was found with source PVC name {src_pvc_name}")
+            return []
```
Comment (PrasadJoshi12) [inline src/oadp_resources/oadp/datamover/volume_snapshot_backup.py:96]:
```suggestion
            return vsb_filtered_list
```

Comment (amastbau) [reply src/oadp_resources/oadp/datamover/volume_snapshot_backup.py:96]: +1

File: src/oadp_resources/velero/backup.py [hunk 3]
```diff
@@ -84,6 +86,10 @@ def failed(self):
     def deleting(self):
         return check_phase(self, self.BackupPhase.DELETING.value)
 
+    def done(self):
+        return self.failed_validation() or self.uploading_partial_failure() or \
```
Comment (PrasadJoshi12) [inline src/oadp_resources/velero/backup.py:90]: Here I think we should check its not in new, Inprogress phase. 
Later we can add the new phase waitingForPlugin phase. 

Let me know what you think ?

Comment (amastbau) [reply src/oadp_resources/velero/backup.py:90]: @sbahar619  @PrasadJoshi12 
Lets go over the new flow together.
?

File: src/oadp_resources/oadp/datamover/volume_snapshot_backup.py (hunk 2 above)
Comment (sbahar619) [reply src/oadp_resources/oadp/datamover/volume_snapshot_backup.py:96]: True, patching...

File: src/oadp_resources/velero/backup.py (hunk 3 above)
Comment (sbahar619) [reply src/oadp_resources/velero/backup.py:90]: Yea you are right, thanks!

**Pull request merged**